frame/dapps-staking/src/pallet/mod.rs

```diff
@@ -0,0 +1,846 @@
+//! Dapps staking FRAME Pallet.
+
+use super::*;
+use frame_support::{
+    dispatch::DispatchResult,
+    ensure,
+    pallet_prelude::*,
+    traits::{Currency, Get, LockIdentifier, LockableCurrency, WithdrawReasons},
+    weights::Weight,
+};
+use frame_system::{ensure_root, ensure_signed, pallet_prelude::*};
+use sp_runtime::{
+    print,
+    traits::{Saturating, Zero},
+    Perbill,
+};
+use sp_std::convert::From;
+
+const STAKING_ID: LockIdentifier = *b"dapstake";
+
+#[frame_support::pallet]
+pub mod pallet {
+    use super::*;
+
+    /// The balance type of this pallet.
+    pub type BalanceOf<T> =
+        <<T as Config>::Currency as Currency<<T as frame_system::Config>::AccountId>>::Balance;
+
+    #[pallet::pallet]
+    #[pallet::generate_store(pub(crate) trait Store)]
+    pub struct Pallet<T>(PhantomData<T>);
+
+    #[pallet::config]
+    pub trait Config: frame_system::Config {
+        /// The staking balance.
+        type Currency: LockableCurrency<Self::AccountId, Moment = Self::BlockNumber>;
+
+        /// Reword amount per block. Will be divided by DAppsRewardPercentage
```

Comment:
Better don’t give pallet that knowledge, let they think that get them all (rewards).

Reply:
I guess this will be removed if we use OnUnbalanced trait to handle rewards